whipper/common/mbngs.py

```diff
@@ -161,10 +161,78 @@ def _getWorks(recording):
     if 'work-relation-list' in recording:
         for work in recording['work-relation-list']:
             if work['type-id'] in valid_work_rel_types:
-                works.append(work['work']['id'])
+                works.append(work['work'])
     return works
 
 
+def _getWorksId(works):
+    """Get work ids from "performance of" works"""
+    ids = []
+    for work in works:
+        ids.append(work['id'])
+    return ids
+
+
+def _getComposers(works):
+    """
+    Extract composer(s) from works' artist-relation-list into a list
+
+    :rtype: list
+    """
+    composers = []
+    valid_artist_rel_type = [
+        'd59d99ea-23d4-4a80-b066-edca32ee158f',  # "Composer"
+    ]
+    for work in works:
+        if 'artist-relation-list' in work:
+            for artist_relation in work['artist-relation-list']:
+                if artist_relation['type-id'] in valid_artist_rel_type:
+                    composerName = artist_relation['artist']['name']
+                    composers.append(composerName)
+    return composers
```

Comment:
Just curious, how does this handle when a single composer is given multiple times? (Either for the same work or for Recordings with multiple Works by the same composer.)

If you test this and it includes the same name multiple times, maybe pass it through a filter to remove duplicates. (Or store it as a set? I don’t know how mutagen will handle being fed a set, so alternatively store as set internally in the function and type case to list when `return`ing.)

Of course, if you test this and it only adds the same name twice, no need to make additional code changes here. :)

Sample recordings for testing: https://musicbrainz.org/recording/69f6f250-e26f-4ce5-95a1-12129671ed7a (same Work two times (I’ll try and remember to fix this later ;)), https://musicbrainz.org/recording/9ea4d8d9-b07e-4360-b24b-f2d405403002 (two different Works, both by same composer).

Reply:
For both cases you gave me, it all returns `['Rune Cygan Barslund', 'Rune Cygan Barslund']` and `['Calvin Harris', 'Calvin Harris']` but i just found out `['[traditional]']` is also a possibility -.-

Reply:
Fixed my previous comment. Did you try out how mutagen acts if given a set instead of a list? Did you look over list methods if there’s maybe something there to clear out duplicates? Otherwise, how about just converting from set to list during `return`?

Reply:
I tried and `mutagen` doesn't support set. This is the error given when i use it (it works with list) :
 `ValueError: {'a', 'b', 'c'} needs to be str for key 'title' `